api/app/src/routes/medical/document.ts

```diff
@@ -63,32 +67,25 @@ router.post(
 /** ---------------------------------------------------------------------------
  * GET /document
  *
- * Downloads the specified document for the specified patient.
+ * Fetches the document from S3 and sends a presigned URL
  *
- * @param req.query.patientId Patient ID for which to retrieve document metadata.
- * @param req.query.facilityId The facility providing NPI for the document download.
- * @param req.query.location The document URL.
- * @param [req.query.mimeType] The mime type of the document.
- * @param [req.query.fileName] The file name of the document.
- * @return The document content.
+ * @param req.query.fileName The file name of the document in s3.
+ * @return presigned url
  */
 router.get(
-  "/download",
+  "/downloadUrl",
   asyncHandler(async (req: Request, res: Response) => {
     const cxId = getCxIdOrFail(req);
-    const patientId = getFromQueryOrFail("patientId", req);
-    const facilityId = getFromQueryOrFail("facilityId", req);
-
-    const location = getFromQueryOrFail("location", req);
-    const mimeType = getFromQuery("mimeType", req);
-    const fileName = getFromQuery("fileName", req);
-
-    fileName && res.attachment(fileName);
-    mimeType && res.header("Content-Type", mimeType);
+    const hour = 3600;
+    const fileName = getFromQueryOrFail("fileName", req);
 
-    await downloadDocument({ patientId, cxId, facilityId, location, stream: res });
+    const url = await s3client.getSignedUrl("getObject", {
+      Bucket: Config.getMedicalDocumentsBucketName(),
+      Key: createS3FileName(cxId, fileName),
+      Expires: hour,
```

Comment:
I think this is too long. A pre-signed URL is basically open to anyone with access to it (I'd say this is one of the reasons CW does not return a URL but has the expense of transferring all files that go through the network using their servers).

That's another reason why we should not return the URL, but send a redirect to a much shorter lived pre-signed URL (maybe 10-20 seconds).